File: output/GPT4o-Mini/xCodeEval_remove_4000/C++/C/eaaddf5db5734cacd0fce60946a15c35.c
#include <stdio.h>
#include <stdlib.h>
#include <string.h>
#include <stdbool.h>

#define MAXN 200200
#define MOD1 987655723
#define MOD2 987656927

typedef unsigned int ui;
typedef long long ll;

typedef struct {
    ui v;
} Field;

Field createField(Long x) {
    Field f;
    f.v = (ui)(x % MOD1);
    return f;
}

Field addField(Field a, Field b) {
    Field r = a;
    if ((ll)a.v + b.v >= MOD1) r.v += b.v - MOD1;
    else r.v += b.v;
    return r;
}

Field subField(Field a, Field b) {
    Field r = a;
    if (a.v < b.v) r.v -= b.v - MOD1;
    else r.v -= b.v;
    return r;
}

Field mulField(Field a, Field b) {
    Field r;
    r.v = (ui)((ll)a.v * b.v % MOD1);
    return r;
}

Field powField(Field a, ui p) {
    Field r = createField(1);
    Field e = a;
    while (p) {
        if (p & 1) r = mulField(r, e);
        e = mulField(e, e);
        p >>= 1;
    }
    return r;
}

Field invField(Field a) {
    return powField(a, MOD1 - 2);
}

Field h1[MAXN], h2[MAXN];
int st[MAXN], p[MAXN], f[MAXN];
char s[MAXN], t[MAXN];

void buildHash() {
    Field cp = createField(7);
    h1[0] = createField(t[0]);
    for (int i = 1; i < strlen(t); i++) {
        h1[i] = addField(h1[i - 1], mulField(createField(t[i]), cp));
        cp = mulField(cp, createField(7));
    }

    cp = createField(13);
    h2[0] = createField(t[0]);
    for (int i = 1; i < strlen(t); i++) {
        h2[i] = addField(h2[i - 1], mulField(createField(t[i]), cp));
        cp = mulField(cp, createField(13));
    }
}

bool getH(int l, int r, Field *hm1, Field *hm2, int *len) {
    if (l + f[l] - 1 >= r) return false;

    int b = st[l + f[l]];
    int e = st[r - p[r]];

    *len = e - b + 1;

    *hm1 = subField(h1[e], h1[b - 1]);
    *hm1 = mulField(*hm1, invField(powField(createField(7), b)));

    if (p[r] % 2 == 1) {
        (*len)++;
        *hm1 = addField(*hm1, powField(createField(7), e - b + 1));
    }

    if (f[l] % 2 == 1) {
        (*len)++;
        *hm1 = addField(createField(1), mulField(*hm1, createField(7)));
    }

    *hm2 = subField(h2[e], h2[b - 1]);
    *hm2 = mulField(*hm2, invField(powField(createField(13), b)));

    if (p[r] % 2 == 1) {
        *hm2 = addField(*hm2, powField(createField(13), e - b + 1));
    }

    if (f[l] % 2 == 1) {
        *hm2 = addField(createField(1), mulField(*hm2, createField(13)));
    }

    return true;
}

void solve() {
    int n;
    scanf("%d %s", &n, s);
    n = strlen(s);

    s[0] = '0';
    s[n + 1] = '0';
    s[n + 2] = '\0';
    n += 2;

    for (int i = 0; i < n; i++) {
        if (s[i] == '1' && s[i + 1] == '1') {
            i++;
            continue;
        }
        t[strlen(t)] = s[i];
        st[strlen(t) - 1] = strlen(t) - 1;
    }

    p[0] = (s[0] == '1') ? 1 : 0;

    for (int i = 1; i < n; i++) {
        p[i] = (s[i] == '0') ? 0 : p[i - 1] + 1;
    }

    f[n - 1] = 0;

    for (int i = n - 2; i >= 0; i--) {
        f[i] = (s[i] == '0') ? 0 : f[i + 1] + 1;
    }

    buildHash();

    int q;
    scanf("%d", &q);
    for (int i = 0; i < q; i++) {
        int a, b, c;
        scanf("%d %d %d", &a, &b, &c);

        Field hm1, hm2;
        int len;
        if (getH(a, a + c - 1, &hm1, &hm2, &len) && getH(b, b + c - 1, &hm1, &hm2, &len)) {
            printf("Yes\n");
        } else {
            printf("No\n");
        }
    }
}

int main() {
    solve();
    return 0;
}

File: output/GPT4o-Mini/xCodeEval_remove_4000/C++/Java/eaaddf5db5734cacd0fce60946a15c35.java
import java.util.*;
import java.io.*;

class Field<N> {
    private final int MOD;
    private int v;

    public Field(int x, int mod) {
        this.MOD = mod;
        this.v = x % MOD;
    }

    public Field<N> pow(int p) {
        return new Field<>(pow(v, p), MOD);
    }

    private int pow(int a, int p) {
        int r = 1, e = a;
        while (p > 0) {
            if ((p & 1) == 1) {
                r = (int) ((long) r * e % MOD);
            }
            e = (int) ((long) e * e % MOD);
            p >>= 1;
        }
        return r;
    }

    public Field<N> inv() {
        return new Field<>(pow(v, MOD - 2), MOD);
    }

    public Field<N> add(Field<N> o) {
        return new Field<>((v + o.v) % MOD, MOD);
    }

    public Field<N> subtract(Field<N> o) {
        return new Field<>((v - o.v + MOD) % MOD, MOD);
    }

    public Field<N> multiply(Field<N> o) {
        return new Field<>((int) ((long) v * o.v % MOD), MOD);
    }

    public Field<N> divide(Field<N> o) {
        return multiply(o.inv());
    }

    public static Field<Integer>[] fact(int t, int mod) {
        Field<Integer>[] F = new Field[t + 1];
        F[0] = new Field<>(1, mod);
        for (int i = 1; i <= t; i++) {
            F[i] = F[i - 1].multiply(new Field<>(i, mod));
        }
        return F;
    }

    public static Field<Integer>[] invfact(int t, int mod) {
        Field<Integer>[] F = new Field[t + 1];
        F[t] = new Field<>(1, mod);
        Field<Integer> X = new Field<>(1, mod);
        for (int i = 2; i <= t; i++) {
            X = X.multiply(new Field<>(i, mod));
        }
        F[t] = X.inv();
        for (int i = t - 1; i >= 2; i--) {
            F[i] = F[i + 1].multiply(new Field<>(i + 1, mod));
        }
        return F;
    }

    @Override
    public String toString() {
        return String.valueOf(v);
    }
}

class DDostizhimieStroki {
    private static final int MAXN = 200200;
    private int[] st = new int[MAXN];
    private int[] p = new int[MAXN];
    private int[] f = new int[MAXN];
    private String s;
    private StringBuilder t = new StringBuilder();
    private Field<Integer>[] h1 = new Field[MAXN];
    private Field<Integer>[] h2 = new Field[MAXN];
    private Field<Integer> p1 = new Field<>(7, 987655723);
    private Field<Integer> p2 = new Field<>(13, 987656927);

    private void buildHash() {
        Field<Integer> cp = p1;
        h1[0] = new Field<>(t.charAt(0) - '0', 987655723);
        for (int i = 1; i < t.length(); i++) {
            h1[i] = h1[i - 1].add(new Field<>(t.charAt(i) - '0', 987655723).multiply(cp));
            cp = cp.multiply(p1);
        }

        cp = p2;
        h2[0] = new Field<>(t.charAt(0) - '0', 987656927);
        for (int i = 1; i < t.length(); i++) {
            h2[i] = h2[i - 1].add(new Field<>(t.charAt(i) - '0', 987656927).multiply(cp));
            cp = cp.multiply(p2);
        }
    }

    private Pair<Integer, Integer> getH(int l, int r) {
        if (l + f[l] - 1 >= r) return new Pair<>(-1, -1);

        int b = st[l + f[l]];
        int e = st[r - p[r]];

        int len = e - b + 1;

        Field<Integer> hm1 = h1[e].subtract(h1[b - 1]).divide(p1.pow(b));
        if (p[r] % 2 == 1) {
            len++;
            hm1 = hm1.add(p1.pow(e - b + 1));
        }

        if (f[l] % 2 == 1) {
            len++;
            hm1 = new Field<>(1, 987655723).add(hm1.multiply(p1));
        }

        Field<Integer> hm2 = h2[e].subtract(h2[b - 1]).divide(p2.pow(b));
        if (p[r] % 2 == 1) {
            hm2 = hm2.add(p2.pow(e - b + 1));
        }

        if (f[l] % 2 == 1) {
            hm2 = new Field<>(1, 987656927).add(hm2.multiply(p2));
        }

        return new Pair<>(hm1.v, hm2.v);
    }

    private String id6(int l, int r) {
        if (l + f[l] - 1 >= r) return "-1";

        int b = st[l + f[l]];
        int e = st[r - p[r]];

        StringBuilder ans = new StringBuilder(t.substring(b, e + 1));
        if (p[r] % 2 == 1) ans.append('1');
        if (f[l] % 2 == 1) ans.insert(0, '1');
        return ans.toString();
    }

    public void solve(InputStream in, OutputStream out) throws IOException {
        BufferedReader reader = new BufferedReader(new InputStreamReader(in));
        PrintWriter writer = new PrintWriter(out);

        int n = Integer.parseInt(reader.readLine());
        s = reader.readLine();
        n = s.length();

        s = '0' + s + '0';

        for (int i = 0; i < s.length(); i++) {
            if (s.charAt(i) == '1' && s.charAt(i + 1) == '1') {
                i++;
                continue;
            }
            t.append(s.charAt(i));
            st[i] = t.length() - 1;
        }

        p[0] = s.charAt(0) == '1' ? 1 : 0;

        for (int i = 1; i < s.length(); i++) {
            p[i] = s.charAt(i) == '0' ? 0 : p[i - 1] + 1;
        }

        f[s.length() - 1] = 0;

        for (int i = s.length() - 1; i >= 0; i--) {
            f[i] = s.charAt(i) == '0' ? 0 : f[i + 1] + 1;
        }

        buildHash();

        int q = Integer.parseInt(reader.readLine());

        for (int i = 0; i < q; i++) {
            StringTokenizer st = new StringTokenizer(reader.readLine());
            int a = Integer.parseInt(st.nextToken());
            int b = Integer.parseInt(st.nextToken());
            int c = Integer.parseInt(st.nextToken());

            if (getH(a, a + c - 1).equals(getH(b, b + c - 1))) {
                writer.println("Yes");
            } else {
                writer.println("No");
            }
        }

        writer.flush();
    }

    public static void main(String[] args) throws IOException {
        DDostizhimieStroki solver = new DDostizhimieStroki();
        solver.solve(System.in, System.out);
    }
}
